tests/lib/prepare.sh

```diff
@@ -717,6 +844,8 @@ EOF
         setup_core_for_testing_by_modify_writable "$UNPACK_DIR"
     fi
 
+    umount /mnt
```

Comment:
could you add a comment here about why this is needed? I see from the commit message, but it would be good to have here since this bit of code is pretty auto-magical and comments help balance that

Reply:
this is actually what broke UC16 + UC18, this is already done in `setup_core_for_testing_by_modify_writable`, so we failed to do it again

Reply:
I fixed this and added a comment about it